modules/bullet/area_bullet.cpp

```diff
@@ -0,0 +1,281 @@
+
+	if (!areaGodoObject) {
+		event.event_callback_id = 0;
+		return;
+	}
+
+	call_event_res[0] = p_status;
+	call_event_res[1] = p_otherObject->get_self(); // Other body
+	call_event_res[2] = p_otherObject->get_instance_id(); // instance ID
+	call_event_res[3] = 0; // other_body_shape ID
+	call_event_res[4] = 0; // self_shape ID
+
+	Variant::CallError outResp;
+	areaGodoObject->call(
+			event.event_callback_method, (const Variant **)call_event_res_ptr, 5, outResp);
+}
+
+void AreaBullet::scratch() {
+	if (isScratched)
+		return;
+	isScratched = true;
+}
+
+void AreaBullet::remove_all_overlapping_instantly() {
+	CollisionObjectBullet *supportObject;
+	for (int i = overlappingObjects.size() - 1; 0 <= i; --i) {
+		supportObject = overlappingObjects[i].object;
+		call_event(supportObject, PhysicsServer::AREA_BODY_REMOVED);
+		supportObject->on_exit_area(this);
+	}
+	overlappingObjects.clear();
+}
+
+void AreaBullet::remove_overlapping_instantly(CollisionObjectBullet *p_object) {
+	CollisionObjectBullet *supportObject;
+	for (int i = overlappingObjects.size() - 1; 0 <= i; --i) {
+		supportObject = overlappingObjects[i].object;
+		if (supportObject == p_object) {
+			call_event(supportObject, PhysicsServer::AREA_BODY_REMOVED);
+			supportObject->on_exit_area(this);
+			overlappingObjects.remove(i);
+			break;
+		}
+	}
+}
+
+int AreaBullet::find_overlapping_object(CollisionObjectBullet *p_colObj) {
+	const int size = overlappingObjects.size();
+	for (int i = 0; i < size; ++i) {
+		if (overlappingObjects[i].object == p_colObj) {
+			return i;
+		}
+	}
+	return -1;
+}
+
+void AreaBullet::set_monitorable(bool p_monitorable) {
+	monitorable = p_monitorable;
+}
+
+void AreaBullet::reload_body() {
+	if (space) {
+		space->remove_area(this);
+		space->add_area(this);
+	}
+}
+
+void AreaBullet::set_space(SpaceBullet *p_space) {
+	// Clear the old space if there is one
+	if (space) {
+		isScratched = false;
+
+		// Remove this object form the physics world
+		space->remove_area(this);
+	}
+
+	space = p_space;
+
+	if (space) {
+		space->add_area(this);
+	}
+}
+
+void AreaBullet::on_collision_filters_change() {
+	if (space) {
+		space->reload_collision_filters(this);
+	}
+}
+
+void AreaBullet::add_overlap(CollisionObjectBullet *p_otherObject) {
+	scratch();
+	overlappingObjects.push_back(OverlappingObjectData(p_otherObject, OVERLAP_STATE_ENTER));
+	p_otherObject->notify_new_overlap(this);
+}
+
+void AreaBullet::put_overlap_as_exit(int p_index) {
+	scratch();
+	overlappingObjects[p_index].state = OVERLAP_STATE_EXIT;
+}
+
+void AreaBullet::put_overlap_as_inside(int p_index) {
+	// This check is required to be sure this body was inside
+	if (OVERLAP_STATE_DIRTY == overlappingObjects[p_index].state) {
+		overlappingObjects[p_index].state = OVERLAP_STATE_INSIDE;
+	}
+}
+
+void AreaBullet::set_transform(const Transform &p_global_transform) {
+	btTransform btTrans;
+	G_TO_B(p_global_transform.orthonormalized(), btTrans);
```

Comment:
Orthonormalize won't give you the global rotation/orientation of the object. See why, and a counter-example [here](https://godot.eska.me/irc-logs/devel/2017-08-26.log) I gave you 3 days ago.